runtime/lua/vim/filetype.lua

```diff
@@ -2565,6 +2565,15 @@ local function normalize_path(path, as_pattern)
   return normal
 end
 
+local abspath = function(x)
+  return fn.fnamemodify(x, ':p')
+end
+if fn.has('win32') == 1 then
+  abspath = function(x)
+    return (fn.fnamemodify(x, ':p'):gsub('\\', '/'))
```

Comment:
After discussion, since match() is a performance-sensitive call, we decided to define a local `abspath()` which does the minimum work necessary to address the issue.